crates/red_knot_vendored/build.rs

```diff
@@ -55,21 +57,32 @@ fn zip_dir(directory_path: &str, writer: File) -> ZipResult<File> {
         // Some unzip tools unzip files with directory paths correctly, some do not!
         if absolute_path.is_file() {
             println!("adding file {absolute_path:?} as {normalized_relative_path:?} ...");
-            zip.start_file(normalized_relative_path, options)?;
+            zip.start_file(&*normalized_relative_path, options)?;
             let mut f = File::open(absolute_path)?;
             std::io::copy(&mut f, &mut zip).unwrap();
+
+            // Patch the VERSIONS file to make `knot_extensions` available
+            if normalized_relative_path == "stdlib/VERSIONS" {
+                writeln!(&mut zip, "knot_extensions: 3.0-")?;
+            }
         } else if !normalized_relative_path.is_empty() {
             // Only if not root! Avoids path spec / warning
             // and mapname conversion failed error on unzip
             println!("adding dir {absolute_path:?} as {normalized_relative_path:?} ...");
             zip.add_directory(normalized_relative_path, options)?;
         }
     }
+
+    // Patch typeshed and add the stubs for the `knot_extensions` module
+    println!("adding file {KNOT_EXTENSIONS_STUBS} as stdlib/knot_extensions.pyi ...");
+    zip.start_file("stdlib/knot_extensions.pyi", options)?;
+    let mut f = File::open(KNOT_EXTENSIONS_STUBS)?;
+    std::io::copy(&mut f, &mut zip).unwrap();
+
     zip.finish()
 }
 
 fn main() {
-    println!("cargo::rerun-if-changed={TYPESHED_SOURCE_DIR}");
```

Comment:
Removing this means that we will rebuild `red_knot_vendored` if *any* file in the package has changed ([ref](https://doc.rust-lang.org/cargo/reference/build-scripts.html#change-detection)). I need this, because `knot_extensions.pyi` is not in the `vendor/typeshed` directory, and we need to pick up changes to it.

And I don't think there's a huge drawback since there are no files in `red_knot_vendored` that should *not* trigger a rebuild except for the README.